### PR TITLE
WIP: added zlib

### PR DESCRIPTION
**Pull request by groutr**
Add zlib.

### CODE REVIEW

File: recipes/zlib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  version: 1.2.8
+
+source:
+  fn: zlib-1.2.8.tar.gz
+  url: http://zlib.net/zlib-1.2.8.tar.gz
+  md5: 44d667c142d7cda120332623eab69f40
+
+build:
+  features:
+    - vc9     [win and py27]
```
Comment (msarahan) [inline recipes/zlib/meta.yaml]: need # in front of selectors

File: recipes/zlib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+requirements:
+  build:
+    - cmake
+
+about:
+  home: http://zlib.net/
+  license: zlib (http://zlib.net/zlib_license.html)
+  license_family: Other
+  license_file: license.txt
+  summary: massively spiffy yet delicately unobtrusive compression library
```
Comment (msarahan) [inline recipes/zlib/meta.yaml]: Need maintainers.  Can add me if you want.

**Comment by msarahan** [comment]
Build looks OK on Travis - just the license file copying bit isn't quite right.

**Comment by jakirkham** [comment]
Strange they don't include the license with the source code.

File: recipes/zlib/build.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+./configure --shared --prefix=$PREFIX
+make
+make install
```
Comment (jakirkham) [inline recipes/zlib/build.sh]: `make check`?

File: recipes/zlib/bld.bat [hunk 4]
```diff
@@ -0,0 +1,14 @@
+mkdir build
+cd build
+
+REM Configure step
+cmake -G "NMake Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=%LIBRARY_PREFIX% -DCMAKE_INSTALL_PREFIX:PATH=%LIBRARY_PREFIX% %SRC_DIR%
+if errorlevel 1 exit 1
+
+cmake --build . --target INSTALL --config Release
+if errorlevel 1 exit 1
+
```
Comment (jakirkham) [inline recipes/zlib/bld.bat]: `ctest`? May need to enable some options to build the tests.

File: recipes/zlib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+package:
+  name: zlib
+  version: 1.2.8
+
+source:
+  fn: zlib-1.2.8.tar.gz
+  url: http://zlib.net/zlib-1.2.8.tar.gz
```
Comment (jakirkham) [inline recipes/zlib/meta.yaml]: Could use jinja for the version to reduce this redundancy.

**Comment by jakirkham** [comment]
Seems the license copying is causing failures on all CIs.

**Comment by groutr** [comment]
Yeah, I see the failures.  The license is embedded in zlib.h.  In fact the license.txt file is simply a copy of that license from the header.
Apparently, license_file seems to be relative to the $SRC_DIR

**Pull request closed without merging**